examples/ChatQnA/deploy/k8s_getting_started.md

```diff
@@ -0,0 +1,90 @@
+|`kubectl delete deployments --all` |             Deletes all deployments in the current namespace, which effectively removes all the managed pods and associated resources.                |
+|`kubectl get pods -o wide`         |              Retrieves a detailed list of all pods in the current namespace, including additional information like IP addresses and the nodes they are running on.               |
+|`kubectl logs <pod-name>`         |         Fetches the logs generated by a container in a specific pod, useful for debugging and monitoring application behavior.                    |
+|`kubectl get svc`                  |            Lists all services in the current namespace, providing a quick overview of the network services and their status.
+
+#### Create and Set Namespace
+A Kubernetes namespace is a logical division within a cluster that is used to isolate different environments, teams, or projects, allowing for finer control over resources and access management. To create a namespace called `chatqa`, use:
+```bash
+kubectl create ns chatqa
+```
+When deploying resources (like pods, services, etc.) into your specific namespace, use the `--namespace` flag with `kubectl` commands, or specify the namespace in your resource configuration files.
+
+To deploy a pod in the `chatqa` namespace:
+```bash
+kubectl apply -f your-pod-config.yaml --namespace=chatqa
+```
+If you want to avoid specifying the namespace with every command, you can set the default namespace for your current context:
+```bash
+kubectl config set-context --current --namespace=chatqa
+```
+
+### Using Helm Charts to Deploy
+
+**What is Helm?** Helm is a package manager for Kubernetes, similar to how apt is for Ubuntu. It simplifies deploying and managing Kubernetes applications through Helm charts, which are packages of pre-configured Kubernetes resources.
+
+#### Key Components of a Helm Chart
+
+| Component         |Description                                                                                                                                             |
+| ---               | ---                                                                                                                                                    |
+| `Chart.yaml`      | This file contains metadata about the chart such as name, version, and description.                                                                    |
+| `values.yaml`     | Stores configuration values that can be customized depending on the deployment environment. These values override defaults set in the chart templates. |
+| `deployment.yaml` | Part of the templates directory, this file describes how the Kubernetes resources should be deployed, such as Pods and Services.                       |
+
+**Update Dependencies:**
+
+- A script called **./update_dependency.sh** is provided which is used to update chart dependencies, ensuring all nested charts are at their latest versions.
```

Comment:
Unique to OPEA.  Normally one just calls Helm dep command directly.

Reply:
Then we should use the standard tools instead of creating our own mechanisms

Reply:
I'm rather new to Helm, so all of this is very much AFAIK:
* Script is there because OPEA has many of the dependencies locally instead of in a Helm repo
* Normally users would update the repo first, and that would basically do what the script is needed for (refresh the deps to latest, not just minimum version)
* OPEA Helm charts are moving to repo, so I think in future that script is needed only in case when one wants to use local checkout

@poussa ?